src/features/barters/do-fetch-barters.js

```diff
@@ -1,7 +1,12 @@
-import { langCode } from '../../modules/lang-helpers';
+import fetch  from 'cross-fetch';
 
-const doFetchBarters = async () => {
-    const language = await langCode();
+const doFetchBarters = async (language) => {
+    language = await new Promise(resolve => {
+        if (!language) {
+            return resolve('en');
+        }
+        return resolve(language);
+    });
```

Comment:
I don't know this syntax. What is the `Promise` and `resolve` for here?

Reply:
In the prebuild, the language code is passed in as a string. In the live site, other functions can pass the language code in as a promise. This ensures that when the language code is provided as a promise, it resolves. It also provides a default if no language code is passed in.

Reply:
I do get it, but why do we need a Promise in the first place? In prebuild it's a statis string and in live it's also a static string from `i18n.language` (`langCode` is marked as `async` but to my understanding it does nothing to remotely qualify as `async`...)

Reply:
We need a promise precisely because the langCode function is marked as async. Anytime a function is marked as async, it _always_ returns a promise, even if the body of the function doesn't return a promise.

Reply:
`langCode` doesn't need to be async, it does nothing to be qualified marked as async.

Reply:
It doesn't need to be async, I've removed it and fetch works perfectly without Promise